Fix #184: Notify `serverJobAdmins` about any new cron jobs

mridul-netizen opened the pull request:
<!--
  - Thanks for submitting code to Oppiabot! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
Fixes #184: Notify `serverJobAdmins` about any new cron jobs

<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
Added three files which on getting changed should trigger the bot
```
CRON_REGISTRY_FILENAME = 'main_cron.py';
CRON_TEST_FILENAME = 'core/controllers/cron_test.py';
CRON_JOB_FILE = 'core/controllers/cron.py';
```
[Comment Body](https://github.com/mridul-netizen/oppia-bot-test-repo-/pull/11#issuecomment-812905921) that bot makes
![image](https://user-images.githubusercontent.com/60519303/113487920-65669300-94d8-11eb-961a-f99e80ec903b.png)

## Checklist
- [x] I have successfully deployed my own instance of Oppiabot.
  - You can find instructions for doing this [here](https://github.com/oppia/oppiabot/wiki/Deploying-your-own-instance-of-the-oppiabot).
- [x] I have manually tested all the changes made in this PR following the [manual tests matrix](https://github.com/oppia/oppiabot/wiki/Manual-Tests-Matrix).

jameesjohn commented on index.js:
```diff
@@ -95,6 +96,9 @@ const runChecks = async (context, checkEvent) => {
             callable.push(checkPullRequestJobModule.checkForNewJob(context));
             break;
+          case constants.cronJobCheck:
+            this.call.push(checkCronJobModule.checkForNewCronJob(context));
```
This does not look right

mridul-netizen replied:
Corrected it

jameesjohn commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,126 @@
+   * @type {import('probot').Octokit.PullsGetResponse} pullRequest
+   */
+  const pullRequest = context.payload.pull_request;
+  if (!hasDatastoreLabel(pullRequest)) {
```
Why are you checking for this?

mridul-netizen replied:
Sorry added it by mistake, removed it!

jameesjohn commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,126 @@
+const CRON_TEST_FILENAME = 'core/controllers/cron_test.py';
+const CRON_JOB_FILE = 'core/controllers/cron.py';
+
+const cronFileIsChanged = (changedFiles) => {
+  changedFiles.filter((file) => {
+    if (file.filename === CRON_JOB_FILE) {
+      return true;
+    }
+  });
+};
```
I don't think this works as expected.
Have you tested that this works?

mridul-netizen replied:
What could be other possible fix to check that any changes have been made to `CRON_JOB_FILE`?

mridul-netizen replied:
I have one more doubt how is `action_build/index.js`  working
![Screenshot from 2021-03-22 15-36-36](https://user-images.githubusercontent.com/60519303/111973411-769cc080-8b24-11eb-9707-9406da6f6a13.png)
![Screenshot from 2021-03-22 15-36-53](https://user-images.githubusercontent.com/60519303/111973417-78668400-8b24-11eb-9b46-d9a69424307f.png)

Why doesn't it contain the cron-job-check I added?

mridul-netizen replied:
I tried console.log inside the if statement it returns true but it is unable to get the comment body
![Screenshot from 2021-03-22 16-09-28](https://user-images.githubusercontent.com/60519303/111977879-4572bf00-8b29-11eb-822d-50bebe854f97.png)

jameesjohn reviewed (changes requested):
Thanks, @mridul-netizen.
I left some comments and will continue reviewing after those have been addressed.

Also, you need to add a link to a PR where this feature has been tested.

mridul-netizen commented on constants.js:
```diff
@@ -110,6 +114,44 @@ const checksWhitelist = {
     [issuesAssignedEvent]: [],
     [pushEvent]: []
   },
+  'oppia-bot-test-repo-': {
+    [openEvent]: [
+      claCheck,
+      changelogCheck,
+      codeOwnerCheck,
+      branchCheck,
+      wipCheck,
+      jobCheck,
+      modelCheck,
+      prTemplateCheck
+    ],
+    [reopenEvent]: [
+      changelogCheck,
+      branchCheck,
+      wipCheck,
+      jobCheck,
+      modelCheck,
+      prTemplateCheck
+    ],
+    [PRLabelEvent]: [assigneeCheck, prLabelCheck],
+    [synchronizeEvent]: [
+      mergeConflictCheck,
+      jobCheck,
+      modelCheck,
+      codeOwnerCheck,
+      oldBuildLabelCheck
+    ],
+    [closeEvent]: [allMergeConflictCheck, updateWithDevelopCheck],
+    [editEvent]: [wipCheck],
+    [issuesLabelEvent]: [issuesLabelCheck],
+    [issuesAssignedEvent]: [issuesAssignedCheck],
+    [unlabelEvent]: [datastoreLabelCheck],
+    [pushEvent]: [forcePushCheck],
+    [periodicCheckEvent]: [periodicCheck],
```
Added this to test actions on my test-repo here https://github.com/mridul-netizen/oppia-bot-test-repo-/pull/8 would remove it at end

mridul-netizen commented:
PTAL @jameesjohn

jameesjohn commented on constants.js:
```diff
@@ -110,6 +114,47 @@ const checksWhitelist = {
     [issuesAssignedEvent]: [],
     [pushEvent]: []
   },
+  'oppia-bot-test-repo-': {
```
This should not be committed.

jameesjohn commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,116 @@
+const CRON_TEST_FILENAME = 'core/controllers/cron_test.py';
+const CRON_JOB_FILE = 'core/controllers/cron.py';
+
+const cronFileIsChanged = (changedFiles) => {
```
Change to `isCronFileChanged`

mridul-netizen replied:
done

jameesjohn commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,116 @@
+   */
+  const pullRequest = context.payload.pull_request;
+  const changedFiles = await getAllChangedFiles(context);
+  const CronFileIsChanged = cronFileIsChanged(changedFiles);
```
Following the earilier suggestion,
```suggestion
  const cronFileIsChanged = cronFileIsChanged(changedFiles);
```

mridul-netizen replied:
done

jameesjohn commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,116 @@
+
+const cronFileIsChanged = (changedFiles) => {
+  let result;
+  changedFiles.filter((file) => {
+    if (file.filename === CRON_JOB_FILE) {
+      result = true;
+    }
+  });
+  return result;
```
Should this function return a boolean?

mridul-netizen replied:
yes

jameesjohn commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,116 @@
+  const newLineFeed = '<br>';
+  const serverJobsForm = 'server jobs form'.link(
+    'https://goo.gl/forms/XIj00RJ2h5L55XzU2');
+  const wikiLinkText = (
+    'this guide'.link(
+      'https://github.com/oppia/oppia/wiki/Running-jobs-in-production' +
+      '#submitting-a-pr-with-a-new-job'));
```
```suggestion
  const wikiLinkText = (
    'this guide'.link(
      'https://github.com/oppia/oppia/wiki/Running-jobs-in-production' +
      '#submitting-a-pr-with-a-new-job'
    )
  );
```

jameesjohn replied:
Also, is the appropriate link for cron jobs? /cc @vojtechjelinek 
This is also used for one-off jobs.

vojtechjelinek replied:
Yes

vojtechjelinek replied:
But this link should preferably be added to constants.

mridul-netizen replied:
done

jameesjohn reviewed (changes requested):
Thanks, @mridul-netizen, I left some comments.

mridul-netizen commented:
@jameesjohn PTAL

vojtechjelinek commented:
@mridul-netizen Some of the comments from James are not addressed.

mridul-netizen commented:
@jameesjohn I addressed all your comments PTAL

jameesjohn commented on spec/checkNewCronJobSpec.js:
```diff
@@ -0,0 +1,444 @@
+
+    app = robot.load(oppiaBot);
+    spyOn(app, 'auth').and.resolveTo(github);
+    spyOn(checkCronJobModule, 'checkForNewCronJob').and.callThrough();
+    spyOn(checkPullRequestBranchModule, 'checkBranch').and.callFake(() => { });
+    spyOn(checkWIPModule, 'checkWIP').and.callFake(() => { });
+    spyOn(
+      checkPullRequestTemplateModule, 'checkTemplate'
+    ).and.callFake(() => { });
+    spyOn(newCodeOwnerModule, 'checkForNewCodeowner').and.callFake(() => { });
```
You need to mock all the modules that listen on the open event. Eg ApiForSheets, pullRequestJobCheck, etc.

mridul-netizen replied:
done

jameesjohn commented on spec/checkNewCronJobSpec.js:
```diff
@@ -0,0 +1,444 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
```
Are you running the tests locally, a lot of them seem to be failing?

mridul-netizen commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,117 @@
+   */
+  const pullRequest = context.payload.pull_request;
+  const changedFiles = await getAllChangedFiles(context);
+  console.log(changedFiles);
```
removed it

mridul-netizen commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,117 @@
+
+const hasNotAddedTests = (changedFiles) => {
+  let result;
+  changedFiles.filter((file) => {
+    if ((file.filename === CRON_REGISTRY_FILENAME ||
+       file.filename === CRON_TEST_FILENAME)) {
+      result = true;
+    }
+  });
```
done

mridul-netizen commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,117 @@
+
+const isCronFileChanged = (changedFiles) => {
+  let result;
+  changedFiles.filter((file) => {
+    if (file.filename === CRON_JOB_FILE) {
+      result = true;
+    }
+  });
```
done

mridul-netizen commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,117 @@
+const CRON_REGISTRY_FILENAME = 'main_cron.py';
+const CRON_TEST_FILENAME = 'core/controllers/cron_test.py';
+const CRON_JOB_FILE = 'core/controllers/cron.py';
+
+const isCronFileChanged = (changedFiles) => {
```
done

mridul-netizen commented:
Hi @jameesjohn thanks for reviewing, I have addressed your comments, the tests are passing with just one fail
![image](https://user-images.githubusercontent.com/60519303/116099194-5931b800-a6c9-11eb-93e8-6d3eb07db043.png)
![image](https://user-images.githubusercontent.com/60519303/116099246-66e73d80-a6c9-11eb-857e-be41c84e2c23.png)
Can you PTAL if the fail is due to my changes?

jameesjohn commented on spec/checkPullRequestJobSpec.js:
```diff
@@ -459,6 +460,7 @@ describe('Pull Request Job Spec', () => {
       checkCriticalPullRequestModule, 'checkIfPRAffectsDatastoreLayer'
     ).and.callFake(() => { });
     spyOn(checkPullRequestBranchModule, 'checkBranch').and.callFake(() => { });
+    spyOn(checkCronJobModule, 'checkForNewCronJob').and.callThrough();
```
```suggestion
    spyOn(checkCronJobModule, 'checkForNewCronJob').and.callFake(() => {});
```

mridul-netizen replied:
done

jameesjohn replied:
Should be callFake not callThrough

jameesjohn commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,121 @@
+ */
+const hasAddedTests = (changedFiles) => {
+  let result;
+  changedFiles.some((file) => {
```
You should `return`  the result of `changedFiles.some`. Currently, you are returning undefined.
Do this here and elsewhere.

mridul-netizen replied:
done

jameesjohn reviewed (changes requested):
Thanks, left some comments.
PTAL @vojtechjelinek.

vojtechjelinek commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,111 @@
+const CRON_REGISTRY_FILENAME = 'main_cron.py';
+const CRON_TEST_FILENAME = 'core/controllers/cron_test.py';
+const CRON_JOB_FILE = 'core/controllers/cron.py';
+const WIKI_LINK = 'https://github.com/oppia/oppia/wiki/' +
+ 'Testing-jobs-and-other-features-on-production';
```
Could we integrate this link across the whole codebase? Since I guess this should also be used for normal jobs.

mridul-netizen replied:
Added to constants.js

vojtechjelinek commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,111 @@
+const WIKI_LINK = 'https://github.com/oppia/oppia/wiki/' +
+ 'Testing-jobs-and-other-features-on-production';
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem[]} changedFiles
+ */
```
Please add proper docstrings.

vojtechjelinek replied:
Ditto in other places.

mridul-netizen replied:
Are these wrong or do I need to add more?I think I covered all of them

mridul-netizen replied:
done

vojtechjelinek commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,111 @@
+ * @param {import('probot').Context} context
+ */
+const checkForNewCronJob = async (context) => {
+  /**
+   * @type {import('probot').Octokit.PullsGetResponse} pullRequest
+   */
```
Why is this here?

jameesjohn replied:
I think it's for type inference.

mridul-netizen replied:
Other files also used the same pattern so I added it.It specifies the type here.

vojtechjelinek commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,111 @@
+const getRegistryReminder = (changedFiles) => {
+  let registryReminder = '';
+  if (!hasAddedTests(changedFiles)) {
+    registryReminder +=
+      ' please add the new test and URL redirects for new cron jobs';
```
```suggestion
    registryReminder +=
      ' please add the new test and URL redirects for the new CRON jobs';
```

vojtechjelinek replied:
Also, this part is pretty cool.

mridul-netizen replied:
done

vojtechjelinek commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,111 @@
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem[]} changedFiles
+ */
+const hasAddedTests = (changedFiles) => {
+  const file = (fileObject) =>
+    fileObject.filename === (CRON_REGISTRY_FILENAME || CRON_TEST_FILENAME);
+  return changedFiles.some(file);
+};
```
If this function is named `hasAddedTests` why do we check for CRON registry?

mridul-netizen replied:
Basically the function checks if there is some changes made within two files i.e. `main_cron.py` for URL redirects and `cron_test.py` for tests ,if this returns true then it does not add the `registryReminder` msg in the comment  and if returned false then adds the msg to the comment. Should I rename this function to something else?

vojtechjelinek replied:
Yep, the name doesn't really make sense.

vojtechjelinek reviewed (changes requested):
Thanks! Left a few comments.

jameesjohn commented on spec/checkNewCronJobSpec.js:
```diff
@@ -0,0 +1,416 @@
+    spyOn(app, 'auth').and.resolveTo(github);
+    spyOn(checkCronJobModule, 'checkForNewCronJob').and.callThrough();
+    spyOn(
+      checkPullRequestJobModule, 'checkForNewJob').and.callThrough(() => { });
```
```suggestion
      checkPullRequestJobModule, 'checkForNewJob').and.callFake(() => { });
```

mridul-netizen replied:
done

jameesjohn commented on spec/checkNewCronJobSpec.js:
```diff
@@ -0,0 +1,416 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
```
Add a test case where the changed files include the cron test file

mridul-netizen replied:
done

jameesjohn reviewed (changes requested):
Thanks, final comments.

mridul-netizen commented:
Thanks for the review @vojtechjelinek @jameesjohn .I have addressed your comments and made the changes.PTAL!

vojtechjelinek commented:
@mridul-netizen Hey, if you want a review please do not forget to add me as an assignee.

mridul-netizen commented:
> @mridul-netizen Hey, if you want a review please do not forget to add me as an assignee.

@vojtechjelinek yes I want a re-review but I don't have enough permissions to assign you as a reviewer here

jameesjohn reviewed (approved):
LGTM!
Thanks, @mridul-netizen!

vojtechjelinek commented on spec/checkPullRequestJobSpec.js:
```diff
@@ -27,6 +27,9 @@ const checkPullRequestTemplateModule =
 const scheduler = require('../lib/scheduler');
+const checkCronJobModule = require('../lib/checkNewCronJobs');
+const constants = require('../constants');
+const WIKI_LINK = constants.getWikiLink();
```
Why do we need to redeclare this here?

mridul-netizen replied:
Link is also used in the tests for checkPullRequestJob ,so we have to call it from constants.js file

vojtechjelinek replied:
I still don't get what you mean by this.

mridul-netizen replied:
Updated some chages PTAL

vojtechjelinek reviewed (changes requested):
Thanks! Few remaining notes.

mridul-netizen commented on constants.js:
```diff
@@ -170,3 +177,7 @@ module.exports.getBlacklistedAuthors = function() {
   return checksWhitelist;
 };
+
+module.exports.getWikiLink = function() {
+  return WIKI_LINK;
+};
```
@vojtechjelinek This wiki link is used by four files 
1. checkForNewCronJob
2. checkForNewCronJobSpec
3. checkPullrequestJob
4. checkPullRequestJobSpec

So to not declare it manually everywhere I'm just calling it from here

mridul-netizen replied:
calling link into 4 files via a function
1. checkForNewCronJob
2. checkForNewCronJobSpec
3. checkPullRequestJob
4. checkPullRequestJobSpec

vojtechjelinek replied:
Okay, but I still don't understand why we need this as a function?

mridul-netizen replied:
Updated some chages PTAL

mridul-netizen commented on constants.js:
```diff
@@ -123,6 +127,8 @@ const checksWhitelist = {
 
 const blacklistedAuthors = ['translatewiki'];
+const WIKI_LINK = 'https://github.com/oppia/oppia/wiki/' +
+ 'Testing-jobs-and-other-features-on-production';
```
done

mridul-netizen commented:
@vojtechjelinek Thanks for the review, I have adressed your comments PTAL

mridul-netizen commented:
@vojtechjelinek I moved the the link to `utils.js`. I think it makes more sense to place such links there.No extra function is required now.PTAL!

vojtechjelinek commented on lib/utils.js:
```diff
@@ -26,6 +26,10 @@ const LABELS_EXCLUDED_FROM_CODEOWNER_ASSIGNMENT = [
 const CODE_OWNERS_FILE_URL =
   'https://raw.githubusercontent.com/oppia/oppia/develop/.github/CODEOWNERS';
+const JOBS_AND_FETURES_TESTING_WIKI_LINK = (
+  'https://github.com/oppia/oppia/wiki/' +
+    'Testing-jobs-and-other-features-on-production'
```
```suggestion
  'https://github.com/oppia/oppia/wiki/' +
  'Testing-jobs-and-other-features-on-production'
```

mridul-netizen replied:
done

vojtechjelinek commented on spec/checkPullRequestJobSpec.js:
```diff
@@ -495,11 +499,8 @@ describe('Pull Request Job Spec', () => {
-          'https://github.com/oppia/oppia/wiki/Running-jobs-in-production' +
-          '#submitting-a-pr-with-a-new-job')
-      );
+      const wikiLinkText =
+       ('this guide'.link(JOBS_AND_FETURES_TESTING_WIKI_LINK));
```
```suggestion
      const wikiLinkText = 'this guide'.link(
        JOBS_AND_FETURES_TESTING_WIKI_LINK);
```

mridul-netizen replied:
done

vojtechjelinek commented on spec/checkPullRequestJobSpec.js:
```diff
@@ -577,11 +578,8 @@ describe('Pull Request Job Spec', () => {
-          'https://github.com/oppia/oppia/wiki/Running-jobs-in-production' +
-          '#submitting-a-pr-with-a-new-job')
-      );
+      const wikiLinkText =
+       ('this guide'.link(JOBS_AND_FETURES_TESTING_WIKI_LINK));
```
```suggestion
      const wikiLinkText = 'this guide'.link(
        JOBS_AND_FETURES_TESTING_WIKI_LINK);
```

mridul-netizen replied:
done

vojtechjelinek commented on spec/checkPullRequestJobSpec.js:
```diff
@@ -662,11 +660,8 @@ describe('Pull Request Job Spec', () => {
-          'https://github.com/oppia/oppia/wiki/Running-jobs-in-production' +
-            '#submitting-a-pr-with-a-new-job')
-      );
+      const wikiLinkText =
+       ('this guide'.link(JOBS_AND_FETURES_TESTING_WIKI_LINK));
```
```suggestion
      const wikiLinkText = 'this guide'.link(
        JOBS_AND_FETURES_TESTING_WIKI_LINK);
```

mridul-netizen replied:
done

vojtechjelinek commented on spec/checkPullRequestJobSpec.js:
```diff
@@ -737,11 +732,8 @@ describe('Pull Request Job Spec', () => {
-          'https://github.com/oppia/oppia/wiki/Running-jobs-in-production' +
-          '#submitting-a-pr-with-a-new-job')
-      );
+      const wikiLinkText =
+       ('this guide'.link(JOBS_AND_FETURES_TESTING_WIKI_LINK));
```
```suggestion
      const wikiLinkText = 'this guide'.link(
        JOBS_AND_FETURES_TESTING_WIKI_LINK);
```

mridul-netizen replied:
done

vojtechjelinek commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,126 @@
+  const newLineFeed = '<br>';
+  const serverJobsForm = 'server jobs form'.link(
+    'https://goo.gl/forms/XIj00RJ2h5L55XzU2');
+  const wikiLinkText = (
+    'this guide'.link(JOBS_AND_FETURES_TESTING_WIKI_LINK));
```
```suggestion
  const wikiLinkText = 'this guide'.link(
    JOBS_AND_FETURES_TESTING_WIKI_LINK);
```

mridul-netizen replied:
done

vojtechjelinek commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,126 @@
+          'if so please request a testing of this CRON job ' +
+          'with this ' + serverJobsForm +
+          ' Please refer to ' + wikiLinkText + ' for details.';
+  return (serverAdminPing + newLineFeed + prAuthorPing +
+          newLineFeed + 'Thanks!');
```
```suggestion
  return (
    serverAdminPing + newLineFeed + prAuthorPing + newLineFeed + 'Thanks!'
  );
```

mridul-netizen replied:
done

vojtechjelinek commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,126 @@
+          ', PTAL at this PR, ' + 'it adds a new cron job.'
+  );
+
+  const prAuthorPing = 'Also @' + prAuthor + registryReminder +
+          ' It seems you have added or edited a CRON job,' +
+          'if so please request a testing of this CRON job ' +
+          'with this ' + serverJobsForm +
+          ' Please refer to ' + wikiLinkText + ' for details.';
```
```suggestion
  const prAuthorPing =  (
    'Also @' + prAuthor + registryReminder + '. It seems you have added or ' + 
    'edited a CRON job, if so please request a testing of this CRON job ' +
    'with this ' + serverJobsForm + ' Please refer to ' + wikiLinkText + ' '
    'for details.'
  );
```

mridul-netizen replied:
done

vojtechjelinek commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,126 @@
+    'this guide'.link(JOBS_AND_FETURES_TESTING_WIKI_LINK));
+
+  const serverAdminPing = (
+    'Hi @' + serverJobAdmins.join(', @') +
+          ', PTAL at this PR, ' + 'it adds a new cron job.'
```
```suggestion
    'Hi @' + serverJobAdmins.join(', @') +
    ', PTAL at this PR, ' + 'it adds a new cron job.'
```

mridul-netizen replied:
done

vojtechjelinek commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,126 @@
+const CRON_JOB_FILE = 'core/controllers/cron.py';
+
+/**
+ * This function checks if the author has made some changes in the
+ *  cron.py file.
```
```suggestion
 * This function checks if the author has made some changes in the
 * cron.py file.
```

mridul-netizen replied:
done

vojtechjelinek commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,126 @@
+};
+
+/**
+ *This function checks if the author has added respective tests
+ *and URL redirects for the new CRON job.
```
```suggestion
 * This function checks if the author has added respective tests
 * and URL redirects for the new CRON job.
```

mridul-netizen replied:
done

vojtechjelinek commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,126 @@
+    'Also @' + prAuthor + registryReminder + ' It seems you have added or ' +
+    'edited a CRON job, if so please request a testing of this CRON job ' +
+    'with this ' + serverJobsForm + ' Please refer to ' + wikiLinkText +
+    ' for details.');
```
```suggestion
    ' for details.'
  );
```

mridul-netizen replied:
done

vojtechjelinek commented on lib/checkNewCronJobs.js:
```diff
@@ -0,0 +1,126 @@
+ * @param {import('probot').Octokit.PullsListFilesResponseItem[]} changedFiles
+ */
+const hasAddedTests = (changedFiles) => {
+  const file = (fileObject) =>
+    (fileObject.filename === CRON_REGISTRY_FILENAME) ||
+    (fileObject.filename === CRON_TEST_FILENAME);
```
```suggestion
  const file = (fileObject) => (
    fileObject.filename === CRON_REGISTRY_FILENAME ||
    fileObject.filename === CRON_TEST_FILENAME
  );
```

mridul-netizen replied:
done

vojtechjelinek reviewed (approved):
Thanks! Few last details.

vojtechjelinek commented:
@mridul-netizen Thanks!
@jameesjohn Can you merge this, please?
